### DISCUSSION TITLE
Errors and difficulties encountred when trying to fit TESS light curve data with PHOEBE

### DISCUSSION THREAD

**Discussion by aydee218** [phoebe-project/phoebe2#714] [Questions]
I'm just getting started with PHEOBE and I found some problems i can't quite find the solution to

First of all my TESS lc data have fluxes in 'count' not 'W/m2' and how can i change the flux type from energy flux/irradiance to count ? is there a way to use normalized flux instead?
the only parameter i change is the period of the eclipse 

Then, I try to fit my light curve. 
When i use the run_solver, it returns a "typeError : TwoGaussianModel.ce() missing 2 required positional arguments: 'Aell' and 'phi0'. " what should i do ?
and also it gives me output of "residuals are not finite in the initial point" I already checked an answer saying the number of bins has to be lowered. i didn't try it yet

I also tried the polyfit model and it gives me solutions but i can't run the model "ValueError: failed to pass checks"
The only solver that runs is the ebai_method “knn” but the fit is disastrous. can't even call it a fit. 
I also tried to play with the initial parameters (ecc, per0 .. ) but i still can't get a fit

Another question, is here a way to make the phases from 0 to 1 and not -0.5 0.5 ?

**Comment by kecnry** [comment]
> First of all my TESS lc data have fluxes in 'count' not 'W/m2' and how can i change the flux type from energy flux/irradiance to count ? is there a way to use normalized flux instead?

Technically PHOEBE does not support flux in counts or normalized flux, but the entire scale of the light curve is dictated by the [passband luminosity options](http://phoebe-project.org/docs/latest/tutorials/pblum) - so as long as you don't interpret the luminosity as having physical meaning, you should be able to use those options to scale your synthetic model to your data.

> When i use the run_solver, it returns a "typeError : TwoGaussianModel.ce() missing 2 required positional arguments: 'Aell' and 'phi0'. " what should i do ?

This sounds like a bug or perhaps a version issue.  @gecheline - any ideas?

> and also it gives me output of "residuals are not finite in the initial point" I already checked an answer saying the number of bins has to be lowered. i didn't try it yet

Please do try adjusting the bins and let us know if that works.

> I also tried the polyfit model and it gives me solutions but i can't run the model "ValueError: failed to pass checks"
The only solver that runs is the ebai_method “knn” but the fit is disastrous. can't even call it a fit.

It should also give you a list of reasons why the checks failed - what do those say?  You can also always call [b.run_checks()](http://phoebe-project.org/docs/latest/api/phoebe.frontend.bundle.Bundle.run_checks) manually to see whether you have valid inputs.

> Another question, is here a way to make the phases from 0 to 1 and not -0.5 0.5 ?

In the output plots?  Not directly from phoebe, but you can always handle your own custom phase-folding/plotting outside phoebe (you have direct access to all the arrays that are used to make the plots).


I hope this helps you get started (a little further)!
-Kyle

**Reply by aydee218** [reply]
Thank you for the help!! 
Adjusting the bins solve it all.
However, I m still wondering how flux data in 'electron/s' should be converted/adapted(?) before adding it in the lc dataset (W/m2)?

**Reply by kecnry** [reply]
electron/s and W/m2 aren't directly convertible units, so I think your choices are to either convert outside phoebe before loading your observations or if you don't care about the absolute value/units of luminosity, you can essentially ignore the units on both the flux and pblum parameters.  @aprsa - any other thoughts?

**Reply by aprsa** [reply]
I wouldn't worry about the units; if you set `pblum_mode=dataset-coupled`, normalization to data will be done for you automatically. Whether it is counts/s or W/m2, it doesn't affect the fit: just remember that the units for passband luminosity *match* the units your data are in.
